File: app/src/main/java/com/example/taxconnect/adapter/TransactionAdapter.java
package com.example.taxconnect.adapter;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.taxconnect.R;
import com.example.taxconnect.model.TransactionModel;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class TransactionAdapter extends RecyclerView.Adapter<TransactionAdapter.TransactionViewHolder> {

    private List<TransactionModel> transactions = new ArrayList<>();

    public void setTransactions(List<TransactionModel> transactions) {
        this.transactions = transactions;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public TransactionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_transaction, parent, false);
        return new TransactionViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull TransactionViewHolder holder, int position) {
        TransactionModel transaction = transactions.get(position);
        holder.bind(transaction);
    }

    @Override
    public int getItemCount() {
        return transactions.size();
    }

    static class TransactionViewHolder extends RecyclerView.ViewHolder {
        TextView tvDescription, tvDate, tvAmount, tvStatus;

        public TransactionViewHolder(@NonNull View itemView) {
            super(itemView);
            tvDescription = itemView.findViewById(R.id.tvDescription);
            tvDate = itemView.findViewById(R.id.tvDate);
            tvAmount = itemView.findViewById(R.id.tvAmount);
            tvStatus = itemView.findViewById(R.id.tvStatus);
        }

        public void bind(TransactionModel transaction) {
            tvDescription.setText(transaction.getDescription());
            tvAmount.setText(String.format(Locale.getDefault(), "₹ %.2f", transaction.getAmount()));
            tvStatus.setText(transaction.getStatus());

            if (transaction.getTimestamp() != null) {
                SimpleDateFormat sdf = new SimpleDateFormat("MMM dd, yyyy HH:mm", Locale.getDefault());
                tvDate.setText(sdf.format(transaction.getTimestamp().toDate()));
            } else {
                tvDate.setText("");
            }

            if ("SUCCESS".equalsIgnoreCase(transaction.getStatus())) {
                tvStatus.setBackgroundResource(R.drawable.bg_status_success);
            } else {
                tvStatus.setBackgroundColor(Color.GRAY);
            }
        }
    }
}

File: tax connect/app/src/main/java/com/example/taxconnect/features/wallet/TransactionAdapter.kt
package com.example.taxconnect.features.wallet

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.example.taxconnect.R
import com.example.taxconnect.data.models.TransactionModel
import java.text.SimpleDateFormat
import java.util.*

import android.widget.ImageView
import com.example.taxconnect.core.utils.PaymentUtils

class TransactionAdapter : RecyclerView.Adapter<TransactionAdapter.TransactionViewHolder>() {

    private var transactions: List<TransactionModel> = ArrayList()

    fun setTransactions(transactions: List<TransactionModel>) {
        this.transactions = transactions
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TransactionViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_transaction, parent, false)
        return TransactionViewHolder(view)
    }

    override fun onBindViewHolder(holder: TransactionViewHolder, position: Int) {
        val transaction = transactions[position]
        holder.bind(transaction)
    }

    override fun getItemCount(): Int = transactions.size

    class TransactionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvDescription: TextView = itemView.findViewById(R.id.tvDescription)
        private val tvDate: TextView = itemView.findViewById(R.id.tvDate)
        private val tvAmount: TextView = itemView.findViewById(R.id.tvAmount)
        private val tvStatus: TextView = itemView.findViewById(R.id.tvStatus)
        private val ivIcon: ImageView = itemView.findViewById(R.id.ivTransactionIcon)

        fun bind(transaction: TransactionModel) {
            tvDescription.text = transaction.description ?: "Transaction"
            
            val sdf = SimpleDateFormat("MMM dd, yyyy HH:mm", Locale.getDefault())
            val date = transaction.timestamp?.toDate()
            tvDate.text = if (date != null) sdf.format(date) else ""
            
            tvAmount.text = PaymentUtils.formatAmount(transaction.amount)
            
            tvStatus.text = transaction.status
            
            val isSuccess = transaction.status == "SUCCESS" || transaction.status == "COMPLETED"
            val isFailed = transaction.status == "FAILED" || transaction.status == "REJECTED"
            val isPending = transaction.status == "PENDING"

            val statusColor = when {
                isSuccess -> R.color.emerald_600
                isPending -> R.color.orange_500
                isFailed -> R.color.error
                else -> R.color.text_muted
            }
            tvAmount.setTextColor(ContextCompat.getColor(itemView.context, statusColor))
            
            val statusBg = when {
                isSuccess -> R.drawable.bg_status_success
                isPending -> R.drawable.bg_status_pending
                else -> R.drawable.bg_status_rejected
            }
            tvStatus.setBackgroundResource(statusBg)

            // Set icon based on description or type if available
            val desc = transaction.description?.lowercase() ?: ""
            val iconRes = when {
                desc.contains("deposit") || desc.contains("top up") -> R.drawable.ic_add_circle
                desc.contains("withdraw") -> R.drawable.ic_arrow_upward
                desc.contains("consultation") || desc.contains("service") -> R.drawable.ic_work_material3
                else -> R.drawable.ic_wallet_material3
            }
            ivIcon.setImageResource(iconRes)
        }
    }
}
